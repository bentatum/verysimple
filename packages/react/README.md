# @verysimple/react

This is the `@verysimple/react` package