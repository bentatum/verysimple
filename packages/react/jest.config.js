/** @type {import('jest').Config} */
module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 72,
      functions: 75,
      lines: 75,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
  prettierPath: null,
};
