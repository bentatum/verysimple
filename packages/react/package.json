{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
