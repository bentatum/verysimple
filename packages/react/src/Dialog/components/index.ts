export * from './DialogTitle';
export * from './DialogCloseButton';
