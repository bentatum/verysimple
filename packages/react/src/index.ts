export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
