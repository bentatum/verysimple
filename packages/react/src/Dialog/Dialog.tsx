import { Dialog as DialogPrimitive, DialogPanel, DialogProps } from '@headlessui/react'
import { FC, forwardRef } from 'react'

export const Dialog = forwardRef<HTMLDivElement, DialogProps>(({ open, children, ...props }, ref) => {
  return (
    <DialogPrimitive ref={ref} open={open} as="div" className="relative z-10 focus:outline-none" {...props}>
      <div className="fixed inset-0 z-10 w-screen overflow-y-auto bg-background/20 backdrop-blur-sm">
        <div className="flex min-h-full items-center justify-center p-4">
          <DialogPanel
            transition
            className="w-full max-w-md rounded bg-foreground p-6 backdrop-blur-2xl duration-100 ease-out data-[closed]:transform-[scale(95%)] data-[closed]:opacity-0"
          >
            {children}
          </DialogPanel>
        </div>
      </div>
    </DialogPrimitive>
  )
})

Dialog.displayName = "Dialog";

export default Dialog;