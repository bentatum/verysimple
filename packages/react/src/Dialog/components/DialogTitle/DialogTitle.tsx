import { Dialog as UIDialog } from "@headlessui/react";
import { FC, forwardRef, ReactNode } from "react";
import classNames from "classnames";
import type { $ElementProps } from "@/types";

export type DialogTitleProps = $ElementProps<typeof UIDialog.Title> & {
  icon?: ReactNode;
  loose?: boolean;
};

export const DialogTitle: FC<DialogTitleProps> = forwardRef<
  HTMLHeadingElement,
  DialogTitleProps
>(({ icon, className = "", children, loose, ...props }, ref) => {
  return (
    <UIDialog.Title
      {...props}
      ref={ref}
      className={classNames(
        "font-display font-bold text-2xl pt-6 px-6",
        {
          "flex items-center": Boolean(icon),
          "whitespace-nowrap": !className.match(/whitespace-/),
          "mb-8": !loose,
          "mb-14": loose,
        },
        className
      )}
    >
      <>
        {icon && <div className="mr-3">{icon}</div>}
        {children}
      </>
    </UIDialog.Title>
  );
});

DialogTitle.displayName = "DialogTitle";

export default DialogTitle;
