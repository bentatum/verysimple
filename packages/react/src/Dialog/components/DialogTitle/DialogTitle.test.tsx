import { render, screen } from "@testing-library/react";
import Dialog from "../../Dialog";
import DialogTitle from "./DialogTitle";
import { InformationCircleIcon } from "@heroicons/react/24/solid";
import "@/test/mocks/window/IntersectionObserver";
import { createRef } from "react";

beforeEach(() => {
  // mute focus trap warning
  jest.spyOn(console, "warn").mockImplementation(() => {});
});

describe("DialogTitle", () => {
  test("snapshot", () => {
    const { container } = render(
      <Dialog open onClose={() => {}}>
        <DialogTitle>Test</DialogTitle>
      </Dialog>
    );
    expect(container).toMatchSnapshot();
  });

  test("is heading", () => {
    render(
      <Dialog open onClose={() => {}}>
        <DialogTitle>Test</DialogTitle>
      </Dialog>
    );
    expect(screen.getByRole("heading", { name: "Test" })).toBeInTheDocument();
  });

  describe("with icon", () => {
    beforeEach(() => {
      render(
        <Dialog open onClose={() => {}}>
          <DialogTitle
            icon={
              <InformationCircleIcon className="h-5 w-5" data-testid="icon" />
            }
          >
            Test
          </DialogTitle>
        </Dialog>
      );
    });
    test("flex", () => {
      expect(screen.getByRole("heading")).toHaveClass("flex", "items-center");
    });
    test("spacing", () => {
      expect(screen.getByTestId("icon").parentElement).toHaveClass("mr-3");
    });
  });

  describe("appearance", () => {
    let element: HTMLElement;
    beforeEach(() => {
      render(
        <Dialog open onClose={() => {}}>
          <DialogTitle>Test</DialogTitle>
        </Dialog>
      );
      element = screen.getByRole("heading");
    });
    test("margin bottom", () => {
      expect(element).toHaveClass("mb-8");
    });
    it("uses display font", () => {
      expect(element).toHaveClass("font-bold");
    });
    it("is bold", () => {
      expect(element).toHaveClass("font-bold");
    });
    test("font size", () => {
      expect(element).toHaveClass("text-2xl");
    });
    test("no wrap", () => {
      expect(element).toHaveClass("whitespace-nowrap");
    });
  });
  it("forwards the ref", () => {
    const ref = createRef<HTMLHeadingElement>();
    render(
      <Dialog open onClose={() => {}}>
        <DialogTitle ref={ref}>Test</DialogTitle>
      </Dialog>
    );
    expect(ref.current).toBeInTheDocument();
  });
  it("has a display name", () => {
    expect(DialogTitle.displayName).toBe("DialogTitle");
  });
});
