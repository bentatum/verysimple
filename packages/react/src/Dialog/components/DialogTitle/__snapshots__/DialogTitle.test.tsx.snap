// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogTitle snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
